File: modified_scripts/c++/matrix_multiplication/deepseek-coder_1.3b.cpp

int matrix_multiplication(const int* A, const size_t nA, const int m1a,
                         const int *B, const size_t m2b,
                         int* C){
    int sum = 0;
    for(int i = 0; i < nA ; ++i){ 
        for(int j = 0; j < m2b ; ++j){ 
            for(int k = 0; k < m1a ; ++k){ 

File: modified_scripts/r/matrix_multiplication/deepseek-coder_1.3b.r

matrix_multiplication = function(A, B) {
    if (nrow(A)!= ncol(B))  stop("Matrix dimensions are not correct")  
    result = rep(0, dim(A)[1]) 									    
    result = 	apply(A, MARGIN=2 , FUN = function (i){ 
        return(sum((B[i, ]*matrix_partitions)))  
    })            
    return(result)}   
